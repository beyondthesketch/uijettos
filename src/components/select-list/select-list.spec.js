import React from 'react';
import renderer from 'react-test-renderer';

import UijettosSelectList from './select-list';

describe('UijettosSelectList component', () => {
    test('Renders correctly when not supplied any props', () => {
        const tree =
            renderer.create(
                <UijettosSelectList />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied child <li>s', () => {
        const tree =
            renderer.create(
                <UijettosSelectList>
                    <li
                        data-value="foo"
                    >Foo</li>
                    <li
                        data-value="bar"
                    >Bar</li>
                    <li
                        data-value="baz"
                    >Baz</li>
                </UijettosSelectList>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied child <li>s and an initial selected', () => {
        const tree =
            renderer.create(
                <UijettosSelectList
                    initialSelected="bar"
                >
                    <li
                        data-value="foo"
                    >Foo</li>
                    <li
                        data-value="bar"
                    >Bar</li>
                    <li
                        data-value="baz"
                    >Baz</li>
                </UijettosSelectList>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied a cssClassPrefix', () => {
        const tree =
            renderer.create(
                <UijettosSelectList
                    cssClassPrefix="doodads"
                >
                    <li
                        data-value="foo"
                    >Foo</li>
                    <li
                        data-value="bar"
                    >Bar</li>
                    <li
                        data-value="baz"
                    >Baz</li>
                </UijettosSelectList>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    describe('Stateless mode', () => {
        test('Renders correctly when selected prop is set to a value, ignoring the initialSelected', () => {
            const tree = renderer
                .create(
                    <UijettosSelectList
                        initialSelected="bar"
                        selected="baz"
                    >
                        <li
                            data-value="foo"
                        >Foo</li>
                        <li
                            data-value="bar"
                        >Bar</li>
                        <li
                            data-value="baz"
                        >Baz</li>
                    </UijettosSelectList>
                )
                .toJSON();

            expect(tree).toMatchSnapshot();
        })
    })
});