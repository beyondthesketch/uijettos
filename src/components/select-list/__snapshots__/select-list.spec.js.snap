// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UijettosSelectList component Renders correctly when not supplied any props 1`] = `
<div
  className="uijettos-select-list"
>
  <ul
    className="uijettos-select-list__options-list"
    onClick={[Function]}
  />
</div>
`;

exports[`UijettosSelectList component Renders correctly when supplied a cssClassPrefix 1`] = `
<div
  className="doodads-select-list"
>
  <ul
    className="doodads-select-list__options-list"
    data-selected="foo"
    onClick={[Function]}
  >
    <li
      className="doodads-select-list__option doodads-select-list__option--selected"
      data-value="foo"
      onKeyPress={[Function]}
    >
      Foo
    </li>
    <li
      className="doodads-select-list__option"
      data-value="bar"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Bar
    </li>
    <li
      className="doodads-select-list__option"
      data-value="baz"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Baz
    </li>
  </ul>
</div>
`;

exports[`UijettosSelectList component Renders correctly when supplied child <li>s 1`] = `
<div
  className="uijettos-select-list"
>
  <ul
    className="uijettos-select-list__options-list"
    data-selected="foo"
    onClick={[Function]}
  >
    <li
      className="uijettos-select-list__option uijettos-select-list__option--selected"
      data-value="foo"
      onKeyPress={[Function]}
    >
      Foo
    </li>
    <li
      className="uijettos-select-list__option"
      data-value="bar"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Bar
    </li>
    <li
      className="uijettos-select-list__option"
      data-value="baz"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Baz
    </li>
  </ul>
</div>
`;

exports[`UijettosSelectList component Renders correctly when supplied child <li>s and an initial selected 1`] = `
<div
  className="uijettos-select-list"
>
  <ul
    className="uijettos-select-list__options-list"
    data-selected="bar"
    onClick={[Function]}
  >
    <li
      className="uijettos-select-list__option"
      data-value="foo"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Foo
    </li>
    <li
      className="uijettos-select-list__option uijettos-select-list__option--selected"
      data-value="bar"
      onKeyPress={[Function]}
    >
      Bar
    </li>
    <li
      className="uijettos-select-list__option"
      data-value="baz"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Baz
    </li>
  </ul>
</div>
`;

exports[`UijettosSelectList component Stateless mode Renders correctly when selected prop is set to a value, ignoring the initialSelected 1`] = `
<div
  className="uijettos-select-list"
>
  <ul
    className="uijettos-select-list__options-list"
    data-selected="baz"
    onClick={[Function]}
  >
    <li
      className="uijettos-select-list__option"
      data-value="foo"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Foo
    </li>
    <li
      className="uijettos-select-list__option"
      data-value="bar"
      onKeyPress={[Function]}
      tabIndex={0}
    >
      Bar
    </li>
    <li
      className="uijettos-select-list__option uijettos-select-list__option--selected"
      data-value="baz"
      onKeyPress={[Function]}
    >
      Baz
    </li>
  </ul>
</div>
`;
